### PR TITLE
Fixed WebSecurityConfig, UserDetailsServiceImpl and added documentation.

### PR DESCRIPTION
**Pull request by AnastasiaBlack**
Please, take a look at this pull request. I added the changed security files only.

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java [hunk 1]
```diff
@@ -9,49 +11,55 @@
...
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
 
-  @Autowired
-  private UserDetailsServiceImpl userDetailsService;
+    @Autowired
+    private UserDetailsServiceImpl userDetailsService;
+    @Autowired
+    RestAuthenticationEntryPoint restAuthenticationEntryPoint;
+    @Autowired
+    CustomSavedRequestAwareAuthenticationSuccessHandler successHandler;
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java]: Can those fields be private and autowired via constuctor?

File: src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -9,49 +11,55 @@
...
+              .formLogin()
+                .loginPage("/login").permitAll()
+                .successHandler(successHandler)
+                .usernameParameter("username")
+                .passwordParameter("password")
+                .failureHandler(new SimpleUrlAuthenticationFailureHandler()).and()
+              .logout()
+                .logoutUrl("/logout")
+                .logoutSuccessUrl("/").and()
+              .rememberMe().key("token").tokenValiditySeconds(3600);
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java]: Didn't you agree with @andrewkononenko that call indentation should be 4 spaces?

File: src/main/java/com/softserve/teamproject/service/impl/UserDetailsServiceImpl.java [hunk 3]
```diff
@@ -48,9 +46,10 @@ public UserDetails loadUserByUsername(String userName) throws
     if (user == null) {
       throw new UsernameNotFoundException("User " + userName + " is not found.");
     }
+
     Set<GrantedAuthority> grantedAuthorities = new HashSet<>();
-    Role role = user.getRole();
-    grantedAuthorities.add(new SimpleGrantedAuthority(role.getName()));
+    //Role role = user.getRole();
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/service/impl/UserDetailsServiceImpl.java]: Doesn't look to be used any more, so we can remove it, can't we?

File: src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java [hunk 4]
```diff
@@ -9,49 +11,55 @@
...
+        auth.authenticationProvider(authenticationProvider());
+    }
 
+    @Bean
+    public DaoAuthenticationProvider authenticationProvider() {
+        DaoAuthenticationProvider authProvider
+                = new DaoAuthenticationProvider();
+        authProvider.setUserDetailsService(userDetailsService);
+        return authProvider;
+    }
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java]: Can't autowiring handle this?

File: src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java [hunk 5]
```diff
@@ -9,49 +11,55 @@
...
-  @Bean
-  public DaoAuthenticationProvider authenticationProvider() {
-    DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
-    authProvider.setUserDetailsService(userDetailsService);
-    return authProvider;
-  }
+    @Bean
+    public CustomSavedRequestAwareAuthenticationSuccessHandler getSuccessHandler() {
+        return new CustomSavedRequestAwareAuthenticationSuccessHandler();
+    }
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java]: And this?

Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java]: This wasn't working without bean

**Pull request merged**